static/js/jira-manual-app-creation.js

```diff
@@ -1,79 +1,130 @@
+}
+
+const gitHubAppPutRequest = (uuid, isUpdate, data) => {
+	$.ajax({
+		type: "PUT",
+		url: `/jira/connect/enterprise/app/${uuid}`,
+		data,
+		success: function (response) {
+			if (response.success) {
+				AP.history.back();
+			} else {
+				handleFormErrors(isUpdate);
+			}
+		}
+	});
 }
 
+$(document).ready(function() {
+	const update = document.getElementById("Update");
+	const isUpdate = update && update.innerText === "Update";
+
+	// Display the filename and make the data valid so users don't need to upload their pem file every time
+	if (isUpdate) {
+		$(".jiraManualAppCreation__formNoFileUploaded").hide();
+		$(".jiraManualAppCreation__formFileUploaded").css('display', 'flex');
+		$("#privateKeyFile").attr("data-aui-validation-state", "valid");
+	}
+});
+
 AJS.$("#jiraManualAppCreation__form").on("aui-valid-submit", (event) => {
-  event.preventDefault();
-  const form = event.target;
-  const csrf = $("#_csrf").val();
-  const data = $(form).serializeObject();
-  const isUpdate = $('input[type=submit]').val() === "Update";
-
-  // Reading the content of the file
-  const file = $("#privateKeyFile")[0].files[0];
-  const reader = new FileReader();
-  reader.readAsDataURL(file);
-
-  reader.onload = () => {
-    data.privateKey = reader.result;
-
-    AP.context.getToken((token) => {
-      data.jwt = token;
-      data._csrf = csrf;
-      data.jiraHost = jiraHost;
-
-      if (isUpdate) {
-        // TODO: Do a put request to update the existing app
-      } else {
-         $.post("/jira/connect/enterprise/app", data, (response, _status, result) => {
-          if (result.status === 201) {
-            // TODO: This doesn't work, will be done in ARC-1565
-            const child = openChildWindow(`/session/github/${response.data.uuid}/configuration?ghRedirect=to`);
-            child.window.jiraHost = jiraHost;
-            child.window.jwt = token;
-          }
-        });
-      }
-    });
-  };
+	event.preventDefault();
+	const form = event.target;
+	const data = $(form).serializeObject();
+	const updateELement = document.getElementById("Update");
+	const isUpdate = updateELement && update.innerText === "Update";
+	const uuid = $(event.target).data("app-uuid");
+	const appName = $(event.target).data("app-appname");
+	const existingPrivateKey = $(event.target).data("app-privatekey");
+
+	AP.context.getToken((token) => {
+		data.jwt = token;
+		data.jiraHost = jiraHost;
+
+		const renderedFilename = document.getElementById("jiraManualAppCreation__uploadedFile").innerText;
+		const isFileChanged = renderedFilename !== `${appName}.private-key.pem`;
+
+		if (isFileChanged || !isUpdate) {
+			// Reading the content of the file
+			const file = $("#privateKeyFile")[0].files[0];
+			const reader = new FileReader();
+			reader.readAsDataURL(file);
+
+			reader.onload = () => {
+				data.privateKey = reader.result;
+			};
+		} else {
+			data.privateKey = existingPrivateKey;
```

Comment:
For the manual form (non-edit mode) or the editing form when the private key file is changed, we'll set the privateKey to reader.result, otherwise set it to the existingPrivateKey.